kiwi/system/setup.py

```diff
@@ -302,8 +299,10 @@ def setup_users(self):
                 user_id = user.get_id()
                 user_realname = user.get_realname()
                 user_shell = user.get_shell()
-                user_secgroups = user.get_secgroups()
-                user_group = user.get_group()
+                if user.get_groups():
+                    user_groups = user.get_groups().split(',')
+                else:
+                    user_groups = None
```

Comment:
Good point ;) that's why I included the groups_list type in the schema with the appropriate pattern, so I does not allow an empty string thus split won't return an empty list.

However, once again, the good option would be to directly get a validated list from xml_parse, as you said.